lua/orgmode/org/hyperlinks/init.lua

```diff
@@ -216,9 +216,10 @@ end
 function Hyperlinks.insert_link(link_location)
   local selected_link = Link:new(link_location)
```

Comment:
The full url including the description a.k.a. headline title is inserted by the user via autocompletion. Link construction parses it and we can get the parts needed parts easily from it to assemble the actual link.